File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    
    tmp = []
    for idx, i in enumerate(priorities):
        tmp.append((i, idx))

    while tmp:
        current = tmp.pop(0)
        if any(current[0] < x[0] for x in tmp):
            tmp.append(current)
        else:
            answer += 1
            if current[1] == location:
                return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <queue>

using namespace std;

int solution(vector<int> priorities, int location) {
    // 2 1 3 2 => 1 3 2 2 => 3 2 2 1 => 3 
    // 2 2 1
    int answer = 0;
    priority_queue<int> pq;
    queue<pair<int, int>> q;
    for(int i: priorities) {
        pq.push(i);
    }
    for(int i=0; i<priorities.size(); i++) {
        q.push({i, priorities[i]});
    }

    int cnt = 0;
    while(!q.empty()) {
        int cur_index = q.front().first;
        int cur_val = q.front().second;
        int ma = pq.top();
        if (cur_val < ma) {
            q.pop();
            q.push({cur_index, cur_val});
        }
        if (cur_val == ma) {
            cnt++;
            if (cur_index == location) {
                return cnt;
            }
            pq.pop();
            q.pop();
        }
    }
   
    return answer;
}